scripts/analyze

```diff
@@ -0,0 +1,298 @@
+
+def write_modules(list, tree):
+    headers = []
+    inits = {}
+    gbl_inits = {}
+    with open('outdir/include/zjs_modules_gen.h', 'w') as file:
+        for i in list:
+            entry = tree.get(i, None)
+            if entry:
+                if 'init' in entry:
+                    inits[i] = {}
+                    inits[i]["inits"] = entry['init']
+                if 'cleanup' in entry:
+                    inits[i]["cleanups"] = entry['cleanup']
+                if 'header' in entry:
+                    merge_lists(headers, entry['header'])
+                if 'global_init' in entry:
+                    gbl_inits[i] = {}
+                    gbl_inits[i]["inits"] = entry['global_init']
+                if 'global_cleanup' in entry:
+                    gbl_inits[i]["cleanups"] = entry['global_cleanup']
+        # Add include headers
+        file.write("// This is a generated file\n")
+        file.write("// Includes:\n")
+        file.write("#include \"jerryscript.h\"\n")
+        for i in headers:
+            file.write("#include \"%s\"\n" % i)
+        file.write("""typedef jerry_value_t (*initcb_t)();
+typedef void (*cleanupcb_t)();
+typedef struct module {
+    const char *name;
+    initcb_t init;
+    cleanupcb_t cleanup;
+    jerry_value_t instance;
+} module_t;
+typedef struct gbl_module {
+    const char *name;
+    void (*init)();
+    void (*cleanup)();
+} gbl_module_t;
+module_t zjs_modules_array[] = {\n""")
+        for i in inits:
+            file.write("    {\"%s\", %s" % (i, inits[i]["inits"][0]))
+            if 'cleanups' in inits[i]:
+                file.write(", %s" % inits[i]["cleanups"][0])
+            file.write("},\n")
+        file.write("};\n")
+        file.write("gbl_module_t zjs_global_array[] = {\n")
+        for i in gbl_inits:
+            file.write("    {\"%s\", %s" % (i, gbl_inits[i]["inits"][0]))
+            if 'cleanups' in gbl_inits[i]:
+                file.write(", %s" % gbl_inits[i]["cleanups"][0])
+            file.write("},\n")
+        file.write("};\n")
+
+if args['BOARD'] == 'linux':
+    restrict = args['RESTRICT'].split(',')
+    restrict = [i.strip() for i in restrict]
+    build_tree(json_tree, restrict)
+    do_print(json_tree)
+    all_list = []
+    get_dep_list(all_list, args['BOARD'], json_tree)
+    for i in all_list:
+        print("Using module: " + i)
+    write_modules(all_list, json_tree)
+else:
+    if 'JS' in args:
+        with open(args['SCRIPT'], 'r') as s:
+            js = s.read()
+        with open(args['JS'], 'w') as f:
+            for i in os.listdir("modules/"):
+                if re.search(r"require\([\'\"]%s[\"\']\)" % i, js):
+                    js_modules += [i]
+            for i in js_modules:
+                with open('modules/' + i) as mod:
+                    f.write(mod.read())
+                    f.write("\n")
+            f.write(js)
+        js_file = args['JS']
+    else:
+        js_file = args['SCRIPT']
```

Comment:
Diff between JS= and SCRIPT= is unclear here; part of why intent should be documented in comments at the top and/or here.